ext/page_log/palite/palite.cpp

```diff
@@ -26,6 +26,100 @@
  * OTHER DEALINGS IN THE SOFTWARE.
  */
 
+/*-
+ * -= How PALite works =-
+ *
+ * PALite implements PALI (the "Page Log Interface") using SQLite as the
+ * underlying storage engine.
+ *
+ * PALite consists of the three main storage table types:
+ *  - Globals: stores global metadata (last LSN, known table IDs)
+ *  - Checkpoints: stores checkpoint information
+ *  - Pages: stores page data (full pages and deltas)
+ *
+ * Each storage table type (Globals, Checkpoints, Pages) is implemented as a
+ * separate SQLite database file:
+ *  - globals.db
+ *  - checkpoints.db
+ *  - pages_[N].db  - one per WT table (N is the table ID)
+ *
+ * -= Concurrency model =-
+ *
+ * SQLite supports multiple simultaneous read transactions coming from separate
+ * database connections, possibly in separate threads or processes, but only one
+ * simultaneous write transaction.
+ *
+ * As a result, PALite uses a readers-writer lock at the table level to ensure
+ * that read operations can occur concurrently while write operations are
+ * serialized.
+ *
+ * -= Known Limitations =-
+ *
+ * PALite does not currently support multiple processes accessing the same
+ * database files. While SQLite supports this use case, it allows only one
+ * writer at a time. In absence of a proper locking mechanism between multiple
+ * processes at PALite level, multiple writers will conflict and fail.
+ * This limitation may be addressed in future releases.
```

Comment:
How could this limitation be addressed if the main restriction comes from the SQLite side? Do I understand the source of the restriction correctly?